### PR TITLE
Added support for LDAP Server Definitions

### PR DESCRIPTION
**Pull request by andy-maier**
For details, see the commit message.

### CODE REVIEW

File: zhmccli/_cmd_ldap.py [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+import click
+
+import zhmcclient
+from .zhmccli import cli
+from ._helper import print_properties, print_resources, abort_if_false, \
+    options_to_properties, original_options, COMMAND_OPTIONS_METAVAR, \
+    click_exception, add_options, LIST_OPTIONS
+
+
+def find_ldap_srv_def(cmd_ctx, console, ldap_srv_def_name):
```
Comment (EdGue42) [inline zhmccli/_cmd_ldap.py:31]: minor: why abbreviate srv? Makes it just harder to read and pronounce.

Comment (andy-maier) [reply zhmccli/_cmd_ldap.py:31]: The normal name would be `ldap_server_definition` which is 22 charecters long. The naming rules defined in our pylint config file have max lengthts of names of 28..32 depending on the kind of name. I think that 22 is too close to that and limits the ability to express the meaning of the variable. Therefore, I went for an abbreviation. How about `ldapsd`?

File: zhmccli/_cmd_ldap.py [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+
+@cli.group('ldap', options_metavar=COMMAND_OPTIONS_METAVAR)
+def ldap_srv_def_group():
+    """
+    Command group for managing HMC LDAP server definitions.
+
+    In addition to the command-specific options shown in this help text, the
+    general options (see 'zhmc --help') can also be specified right after the
+    'zhmc' command name.
+    """
```
[the remark this answers is not in the document]
Comment (andy-maier) [reply zhmccli/_cmd_ldap.py:51]: No. Click is processing this definition due to the decorator that is used, and its only use is to produce the help text for the command group. The modules for the other resources also don't have a body on the group function.

File: zhmccli/_cmd_ldap.py [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+    cmd_ctx.execute_cmd(lambda: cmd_ldap_srv_def_show(cmd_ctx, ldap_srv_def))
+
+
+@ldap_srv_def_group.command('create', options_metavar=COMMAND_OPTIONS_METAVAR)
+@click.option('--name', type=str, required=True,
+              help='The name of the new LDAP server definition.')
+@click.option('--description', type=str, required=False,
+              help='The description of the new LDAP server definition.')
+@click.option('--primary-hostname-ipaddr', type=str, required=True,
```
[the remark this answers is not in the document]
Comment (andy-maier) [reply zhmccli/_cmd_ldap.py:92]: I was following the property names :-)
but I like `--primary-server` better and have changed it.
Also for `--backup-server`.

File: zhmccli/_cmd_ldap.py (hunk 1 above)
Comment (andy-maier) [reply zhmccli/_cmd_ldap.py]: I have changed:
* any internal names from `ldap_srv_def` to `ldapsd`,
* the metavar of the LDAP Server Definition name in the help text from `LDAP_SRV_DEF` to `LDAP_SERVER_DEFINITION`
* the module name from "ldap.py" to "ldap_server_definition.py"

**Pull request merged**